pyEpiabm/pyEpiabm/sweep/host_progression_sweep.py

```diff
@@ -51,14 +64,16 @@ def __call__(self, time: int):
         :param time: Current simulation time
         :type time: int
         """
-        if not (type(time) == int):
-            raise TypeError('Time needs to be type int')
+
         for cell in self._population.cells:
             for person in cell.persons:
                 if person.time_of_status_change == time:
-                    self._update_next_infection_status(person)
                     person.update_status(person.next_infection_status)
-                    if person.infection_status != \
-                            InfectionStatus.Recovered:
-                        person.time_of_status_change =\
-                            time + self._update_time_to_status_change()
+                    if person.infection_status != InfectionStatus.Recovered:
+                        self._update_next_infection_status(person)
+                        if person.infection_status == InfectionStatus.Exposed:
+                            person.time_of_status_change = time +\
+                                 self._set_latent_time()
+                        else:
+                            person.time_of_status_change = time +\
+                                 self._update_time_to_status_change()
```

Comment:
Open question but it seems to me that this might be better combined into the `_update_time_to_status_change` function, so that the if statement is called within there? That could be tidier/more extendable as we add more transitions

Reply:
Agreed, Will change the methods so that they handle this as opposed to the call function finishing off the calculation